jax/lax/lax.py

```diff
@@ -4878,6 +4903,26 @@ def _cumred_batch_rule(prim, batched_args, batch_dims, *, axis: int):
 batching.primitive_batchers[cumprod_p] = partial(_cumred_batch_rule, cumprod_p)
 
 
+cummax_p = standard_primitive(
```

Comment:
I suspect we can share all of these with `cumprod` too by adding a helper function, perhaps something like this:

```
cummax_p = _cumulative_reductive(lax.max, ...)
```

Reply:
Done.